sdk/app_cpu1/common/drv/amds.c

```diff
@@ -0,0 +1,91 @@
+#include "usr/user_config.h"
+
+#if (USER_CONFIG_ENABLE_AMDS_SUPPORT == 1)
+
+#include "drv/amds.h"
+#include "sys/cmd/cmd_amds.h"
+#include "sys/commands.h"
+#include "sys/defines.h"
+#include "sys/util.h"
+#include "xil_io.h"
+#include <stdbool.h>
+#include <stdint.h>
+
+void amds_init(void)
+{
+    xil_printf("AMDS: Initializing...\r\n");
+
+    cmd_amds_register();
+}
+
+/* Returns a byte where each bit represents the validity of the most recently-returned data
+ * for a single channel on this AMDC. bit[0] is for channel 1, bit[1] is for channel 2...
+ * A '1' bit implies the data is valid, while a '0' bit implies invalid data
+ *
+ * base_addr: AMDS_N_BASE_ADDR, where N is the GPIO port number the AMDS mainboard is connected to
+ *
+ * NOTE: Because the AMDS mainboard firmware does not do anything different for unpopulated
+ *       SensorCards, the validity of those channels is meaningless
+ */
+uint8_t amds_check_data_validity(uint32_t base_addr)
+{
+    return Xil_In8(base_addr + AMDS_CH_VALID_REG_OFFSET);
+}
```

Comment:
Or we could have the user specifiy and port number, and use a case statement to pick the correct base_addr for them. This could be updated in both the AMDS and Eddy Current C code.

Reply:
Although that approach falls apart for the REV D... grr

Reply:
ALTHOUGH, there's already a function that converts a "port" for REV D/E/F to a base addr